packages/react-native-reanimated/src/WorkletEventHandler.ts

```diff
@@ -99,32 +108,74 @@ class WorkletEventHandlerWeb<Event extends object>
     | Record<string, (event: ReanimatedEvent<ReanimatedEvent<Event>>) => void>
     | Record<string, (event: JSEvent<Event>) => void>;
 
+  JSHandlers: Record<string, JSHandler<Event>>; // web JS Handlers need to be merged with Worklet handlers
   worklet: (event: ReanimatedEvent<Event>) => void;
 
   constructor(
     worklet: (event: ReanimatedEvent<Event>) => void,
-    eventNames: string[] = []
+    eventNames: string[] = [],
+    JSHandlers: Record<string, JSHandler<Event>>
   ) {
     this.worklet = worklet;
     this.eventNames = eventNames;
+    this.JSHandlers = JSHandlers;
     this.listeners = {};
     this.setupWebListeners();
   }
 
   setupWebListeners() {
     this.listeners = {};
-    this.eventNames.forEach((eventName) => {
+
+    const eventsFromJSHandlers = Object.keys(this.JSHandlers);
+
+    // events that are both from JS and Worklet handlers
+    const sharedEvents = eventsFromJSHandlers.filter((value) =>
+      this.eventNames.includes(value)
+    );
+
+    // events from only Worklet handlers
+    const restWorkletEvents = this.eventNames.filter(
+      (value) => !eventsFromJSHandlers.includes(value)
+    );
+
+    // events from only JS handlers
+    const restJSEvents = eventsFromJSHandlers.filter(
+      (value) => !this.eventNames.includes(value)
+    );
+
+    // shared events get JS and Worklet handlers merged into a listener to put into a prop
+    sharedEvents.forEach((eventName) => {
+      this.listeners[eventName] = jsListener(
+        eventName,
+        this.worklet,
+        this.JSHandlers[eventName]
+      );
+    });
+
+    // only Worklet events get their own listeners
+    restWorkletEvents.forEach((eventName) => {
       this.listeners[eventName] = jsListener(eventName, this.worklet);
     });
+
+    // only JS events get their own listeners
+    restJSEvents.forEach((eventName) => {
+      this.listeners[eventName] = jsListener(
+        eventName,
+        undefined,
+        this.JSHandlers[eventName]
```

Comment:
Instead of passing `undefined` as a prop is better to have second implementation of `jsListener` method